src/Header/EmployeeAssets.js

```diff
@@ -0,0 +1,206 @@
+import React, { useState, useEffect } from "react";
+
+const EmployeeAssets = () => {
+  const [employees, setEmployees] = useState([]);
+  const [assets, setAssets] = useState([]);
+  const [selectedAsset, setSelectedAsset] = useState("");
+  const [selectedEmployee, setSelectedEmployee] = useState("");
+
+  useEffect(() => {
+    const employeesFromStorage = JSON.parse(
+      localStorage.getItem("empkey") || "[]"
+    );
+    setEmployees(employeesFromStorage);
+
+    const assetsFromStorage = JSON.parse(localStorage.getItem("dbKey") || "[]");
+    setAssets(assetsFromStorage);
+  }, []);
+
+  const handleEmployeeChange = (e) => {
+    setSelectedEmployee(e.target.value);
+  };
+
+  function handleAssetChange(event) {
+    setSelectedAsset(event.target.value);
+  }
+
+  const getLinks = () => {
+    const links = JSON.parse(localStorage.getItem("assetLinks"));
+    const linkDetails = [];
+    Object.keys(links).forEach((employeeId) => {
+      const linkIds = links[employeeId];
+      const employeeDetails = employees.find((x) => x.id === employeeId);
+      if (!employeeDetails) return;
+      linkIds.forEach((linkId) => {
+        const assetDetails = assets.find((y) => y.id === linkId);
+        if (!assetDetails) return;
+        linkDetails.push({
+          employeeId: employeeDetails.id,
+          employeeName: employeeDetails.name,
+          asset: assetDetails.laptop,
+          assetId: assetDetails.id,
+        });
+      });
+    });
+    return linkDetails;
+  };
+
+  // const getLinksv2 = () => {
+  //   const links = JSON.parse(localStorage.getItem("assetLinks"));
+  //   const linkDetails = [];
+  //   Object.keys(links).forEach((employeeId) => {
+  //     const linkIds = links[employeeId];
+  //     const employeeDetails = employees.find((x) => x.id === employeeId);
+  //     const assignedEmployee = Object.keys(links).find((employee) =>
+  //       links[employee].includes(selectedAsset)
+  //     );
+  //     if (!employeeDetails) return;
+  //     const objToPush = {
+  //       employeeId: employeeDetails.id,
+  //       employeeName: employeeDetails.name,
+  //       assets: [],
+  //     };
+  //     linkIds.forEach((linkId) => {
+  //       objToPush.assets.push({
+  //         asset: assets.find((y) => y.id === linkId)?.laptop,
+  //         assetId: assets.find((y) => y.id === linkId)?.id,
+  //       });
+  //       if (linkId === selectedAsset) {
+  //         assignedEmployee = employeeDetails.name;
+  //       }
+  //     });
+  //     linkDetails.push(objToPush);
+  //   });
+  //   return linkDetails;
+  // };
+
+  const [rows, setRows] = useState(getLinks());
+
+  useEffect(() => {
+    if (employees.length > 0 && assets.length > 0) {
+      const links = getLinks();
+      setRows(links);
+    }
+  }, [employees, assets]);
+
+  const addLink = (empId, assetId) => {
+    let items = JSON.parse(localStorage.getItem("assetLinks"));
```

Comment:
in your components you should never access localStorage directly, write a utility function and use that utility function here